src/proxy/inbound.rs

```diff
@@ -262,38 +348,58 @@ impl Inbound {
         }
 
         let start = Instant::now();
-        let hbone_addr = req
-            .uri()
-            .to_string()
-            .as_str()
-            .parse::<SocketAddr>()
-            .map_err(|_| {
-                InboundError(
-                    Error::ConnectAddress(req.uri().to_string()),
-                    StatusCode::BAD_REQUEST,
-                )
-            })?;
 
+        // Extract the host or IP from the authority pseudo-header of the URI
+        let hbone_host = req.uri().host().ok_or_else(|| {
+            InboundError(
+                Error::ConnectAddress(req.uri().to_string()),
+                StatusCode::BAD_REQUEST,
+            )
+        })?;
+
+        let hbone_port = req.uri().port_u16().ok_or_else(|| {
+            InboundError(
+                Error::ConnectAddress(req.uri().to_string()),
+                StatusCode::BAD_REQUEST,
+            )
+        })?;
+
+        let hbone_authority = format!("{}:{}", hbone_host, hbone_port);
+
+        let hbone_addr = match hbone_authority.as_str().parse::<SocketAddr>() {
+            Ok(addr) => HboneAddress::SocketAddr(addr),
+            Err(_) => HboneAddress::SvcHostname(hbone_host.into(), hbone_port),
+        };
+
+        // Get the destination workload information of the destination pods (wds) workload (not destination ztunnel)
         let destination_workload = pi
             .local_workload_information
             .get_workload()
             .await
             // At this point we already fetched the local workload for TLS, so it should be infallible.
             .map_err(InboundError::build(StatusCode::SERVICE_UNAVAILABLE))?;
 
-        // Check the request is allowed
-        Self::validate_destination(&pi.cfg, &pi.state, &conn, &destination_workload, hbone_addr)
-            .await
-            .map_err(InboundError::build(StatusCode::BAD_REQUEST))?;
+        // Check the request is allowed by verifying the connection IP and the hbone address IP are the same
```

Comment:
What is the "connection IP"?

Reply:
This comment isn't correct. By connection IP I was referring to the `conn.dst.IP` and the conditional check [here](https://github.com/istio/ztunnel/pull/1420/files/72c87e0a118f47a30068d348d86096baef8a9d61#diff-0acf95d58db195dc1623fb8d2eface8e582459fdcf4a561c5fce509e46270facR550).